nixos/modules/virtualisation/qemu-vm.nix

```
@@ -542,25 +562,19 @@ in
     ];
 
     virtualisation.qemu.drives = mkMerge [
+      [{
+        file = "$NIX_DISK_IMAGE";
```

I propose to give this a name too:

```suggestion
        name = "root";
        file = "$NIX_DISK_IMAGE";
```

Maybe then we can even change the `type = types.nullOr types.str;` above into just `type = types.str;`?

---

Added.

---

My rationale for preferring this name to be nullable is to reduce the burden on anyone adding drives.  The only purpose of adding a name is so you can use it to look up the device-name later.  I figure most use cases won't care about the device name.